### ISSUE TITLE
[Blocks] Implement other event triggers besides `click` & support event triggers across multiple components

### ISSUE THREAD

**Issue by abidlabs** [gradio-app/gradio#723]
Currently, the only event trigger we support is `click`. 

1. Can we add support additional event triggers? Some of these will be component specific, but something like an "on change" event would be useful for many different components. For example, in my Block below, I would like to run the function any time the input sliders change value

```python
block = gr.Blocks()

with block:
    gr.Markdown("Let's do some kinematics! Choose the speed and angle to see the trajectory.")
    
    with gr.Row():
        speed = gr.inputs.Slider(1, 30, default=25)
        angle = gr.inputs.Slider(0, 90, default=45)
    output = gr.outputs.Image(type="plot")
    btn = gr.Button("Run")
    btn.click(plot, [speed, angle], output)

block.launch()
```

2. It would be nice if I could easily assign a trigger to "if any of these two sliders change" without having to write two separate event trigger lines

3. It would be _really_ nice if I could just say something like "if anything changes in this Block" re-run all of the function(s). I would assume that many of our users want this functionality for their blocks. Could we have a parameter in `gr.Blocks()` like `live=True`. In other words, if you run `gr.Blocks(live=True)`, then you have an automatically responsive Block in which if any of the input components change, all of the functions are re-run and the outputs automatically refresh.

**Comment by pngwn** [comment]
**Proposed Component Events**

| Component | Event | When? | Notes |
|---|---|--|--|
| **Audio** | `change`|  audio source changes |  should this be `on:data`? |
|  | `clear`| audio source is reset to nothing | |
| **Button** | `click` | button is clicked | |
| **Carousel** | `change` | current item changes | |
| **ChatBot** | `message` | chatbot posts a message | |
| **Checkbox** | `change` | checkbox is checked/ unchecked | |
| **CheckboxGroup** | `change` | checkbox is checked/ unchecked | |
| **Column** |  | | static layout component (nyi) | 
| **DataFrame** | `change` | data changes | should this be `on:data`? |
| **Dropdown** | `change` | selected option changes | |
| **File** | `change` | file data changes | should this be `on:data`? |
|  | `clear`| video source is reset to no value | |
| **Grid** |  | | static layout component (nyi) | 
| **HighlightedText** | `change` | content is passed in | output only |
| **HTML** | `change` | content is passed in | output only |
| **JSON** | `change` | content is passed in | output only |
| **Label** | `change` | content is passed in | output only |
| **Number** | `change` | number value changes | |
| **Radio** | `change` | selected option changes | |
| **Row** |  | | static layout component | 
| **Slider** | `change` | current value changes | |
| **Tabs** | `change` | selected tab changes | |
| **Textbox** | `change` | text value changes | |
| **Video** | `change`|  video source changes |  should this be `on:data`? |
|  | `clear`| video source is reset to no value | |
| **Timeseries** | `change` | chart data changes | should this `on:data`? |

**Comment by pngwn** [comment]
I think `change` is better than `data` thinking about it. Data kind of makes sense the first time something is loaded but for edits, it makes less sense.

**Comment by FarukOzderim** [comment]
@pngwn could you elaborate on this, how can user reset the video source?
**video source is reset to no value**

**Comment by pngwn** [comment]
Click a button and clear the value / set it to null. The reason for the separate event from `change` is because you typically won’t want to run a prediction on a null value.

**Comment by aliabid94** [comment]
This is great stuff, love it. 
Shouldn't "submit" apply to chatbot too? When a user presses enter. And "change" if a user types a single letter?

**Comment by abidlabs** [comment]
No because the `chatbot` component only displays messages like this: 
![image](https://user-images.githubusercontent.com/1778297/158446896-6d4ed9fe-3972-4c11-8b98-535f28804936.png)

(Doesn't include the input textbox)

**Comment by pngwn** [comment]
Updated the table and implemented all of the events on the frontend in #807.

We can use the table above as a source of truth for syncing front and backend but also for documentation. We should just keep editing it as anything changes.

**Comment by FarukOzderim** [comment]
> Updated the table and implemented all of the events on the frontend in #807.
> 
> We can use the table above as a source of truth for syncing front and backend but also for documentation. We should just keep editing it as anything changes.

Let's reference the changes if we edit the table in the future as it would make further updates easier.

**Comment by FarukOzderim** [comment]
> @pngwn what is Tabs component? I presume it is the Tab inside the Block? and what would be a Tabs change?

Resolved in #844

**Comment by FarukOzderim** [comment]
Btw WDYT about adding events to Blocks? 
For example a change event at Blocks would affect all components in the Blocks.

**Comment by abidlabs** [comment]
Closed via #844
